### PR TITLE
Refactor main

### PR DESCRIPTION
**Pull request by shaunahu**
Did these changes:
- remove unused notebooks
- add type hinting for each function
- refactor `keywordClassifier` for API call
- the whole pipeline is in `pipeline.py`
- Upload a pretrained model for demo use

Will do:
- add notebooks
- revise es search query script

### CODE REVIEW

File: data_discovery_ai/common/parameters.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "batch": 32,
```
Comment (vietnguyengit) [inline data_discovery_ai/common/parameters.json:19]: can you rename this file to make it be specific please? "parameters" of what, try to name something easy to understand without having to find the usage for context.

File: pipeline.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
+import data_discovery_ai.utils.preprocessor as preprocessor
+import data_discovery_ai.model.keywordModel as model
+import data_discovery_ai.utils.es_connector as connector
+import data_discovery_ai.service.keywordClassifier as keywordClassifier
+import numpy as np
+import json
+import pandas as pd
+
+
+def pipeline():
```
Comment (vietnguyengit) [inline pipeline.py:10]: I think you should need to make this pipeline a class instead of tight coupled sequence under one single giant method, each of the step is a separate method with meaningful names and type hinting and description, e.g step 0 is init() 

when you break the steps to method, try to make it not too broad or atomic small unnecessarily, group related actions together by nature or method signature etc.

then you can have a main method that executes sub methods in order. It will be easier to read and debug/maintain later on. Or even if one day we decide to orchestrate the pipeline with tools such as Prefect, we can easily wrap the sub-methods with `@task` decorator and the main method with `@flow` decorator etc.

do you familiar with writing unit tests? ideally you need test cases to cover each of the step that returning values.

Comment (vietnguyengit) [reply pipeline.py:10]: i don't like seeing too much code being commented out.

File: data_discovery_ai/model/keywordModel.py [hunk 3]
```diff
@@ -45,31 +40,38 @@
...
 
 
-def get_class_weights(Y_train):
+"""
+    Calculate label weights by the frequency of a label appears in all records
+    Input:
+        Y_train: numpy.ndarray. The train set of Y
+    Output:
+        label_weight_dic: Dict[int, float]. The label weights, keys are the indexs of labels and values are the weights.
+"""
```
Comment (vietnguyengit) [inline data_discovery_ai/model/keywordModel.py:49]: these should be inside the method, not above

File: data_discovery_ai/model/keywordModel.py [hunk 4]
```diff
@@ -81,9 +83,42 @@ def focal_loss_fixed(y_true, y_pred):
...
+
+
 def keyword_model(
-    X_train, Y_train, X_test, Y_test, class_weight, dim, n_labels, params
-):
+    X_train: np.ndarray,
+    Y_train: np.ndarray,
+    X_test: np.ndarray,
+    Y_test: np.ndarray,
```
Comment (vietnguyengit) [inline data_discovery_ai/model/keywordModel.py:116]: I dont think this syntax is the ideal naming convention: https://peps.python.org/pep-0008/#function-and-variable-names they should e.g x_train, but do you really need to name it that way? why not call x_train to be `feature_matrix` ?

File: data_discovery_ai/service/keywordClassifier.py [hunk 5]
```diff
@@ -1,14 +1,32 @@
...
+"""
+
+
+def keywordClassifier(trained_model, description):
+    with open("data_discovery_ai/common/parameters.json", "r", encoding="utf-8") as f:
+        params = json.load(f)
+    selected_model = model.load_saved_model(trained_model)
+    description_embedding = preprocessor.get_description_embedding(description)
+    dimension = description_embedding.shape[0]
+    target_X = description_embedding.reshape(1, dimension)
```
Comment (vietnguyengit) [inline data_discovery_ai/service/keywordClassifier.py:22]: have a look with this: https://docs.python.org/3/library/configparser.html

**Review by vietnguyengit — CHANGES_REQUESTED** [review]

File: data_discovery_ai/common/parameters.json (hunk 1 above)
Comment (shaunahu) [reply data_discovery_ai/common/parameters.json:19]: file name changed to `keyword_classification_parameters.ini`

File: pipeline.py (hunk 2 above)
Comment (shaunahu) [reply pipeline.py:10]: > I think you should need to make this pipeline a class instead of tight coupled sequence under one single giant method, each of the step is a separate method with meaningful names and type hinting and description, e.g step 0 is init()
> 
> when you break the steps to method, try to make it not too broad or atomic small unnecessarily, group related actions together by nature or method signature etc.
> 
> then you can have a main method that executes sub methods in order. It will be easier to read and debug/maintain later on. Or even if one day we decide to orchestrate the pipeline with tools such as Prefect, we can easily wrap the sub-methods with `@task` decorator and the main method with `@flow` decorator etc.
> 
> do you familiar with writing unit tests? ideally you need test cases to cover each of the step that returning values.

convert the functions within a class. Please let me know if these works for API use or not.
I'd like to separate the unit test to another PR. Have several tasks to do so will work on the unit test when I have time.


> i don't like seeing too much code being commented out.

Me too, just want to make sure these code can run from your side so provide an example of use.

File: data_discovery_ai/model/keywordModel.py (hunk 3 above)
Comment (shaunahu) [reply data_discovery_ai/model/keywordModel.py:49]: Moved inside the methods, changes applied to all functions.

File: data_discovery_ai/model/keywordModel.py (hunk 4 above)
Comment (shaunahu) [reply data_discovery_ai/model/keywordModel.py:116]: I understand python variable names should avoid starting with uppercase. However, in this context, we use uppercase `X` and `Y` to represent feature matrix and target matrix, following common mathematical representations. Here, lowercase letters like `x` or `y` suggest a single item rather than a matrix. Many ML libraries tutorials also use this representation, like https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression and https://github.com/trent-b/iterative-stratification. In these examples, `y` is lowercase because it represents a 1D array. For our multi-label classification task, however, `Y` is a matrix with dimensions no_of_items x no_of_labels, reflecting multiple labels per item. Let me know if this makes sense to you or I can change the terms from `X_train` to `feature_matrix_train`.

File: data_discovery_ai/service/keywordClassifier.py (hunk 5 above)
Comment (shaunahu) [reply data_discovery_ai/service/keywordClassifier.py:22]: > have a look with this: https://docs.python.org/3/library/configparser.html

changed from json to configparser

**Review by vietnguyengit — APPROVED** [review]
